rasa_nlu/extractors/duckling_extractor.py

```diff
@@ -68,34 +73,27 @@ def pipeline_init(self, language):
                 raise Exception("Duckling error. {}".format(e.message))
 
     def process(self, text, entities):
-        # type: (Text, List[Dict[Text, Any]], Text) -> Dict[Text, Any]
+        # type: (Text, List[Dict[Text, Any]]) -> Dict[Text, Any]
 
+        extracted = []
         if self.duckling is not None:
-            parsed = self.duckling.parse(text)
-            for duckling_match in parsed:
-                for entity in entities:
-                    if entity["start"] == duckling_match["start"] and entity["end"] == duckling_match["end"]:
-                        entity["value"] = duckling_match["value"]["value"]
-                        entity["duckling"] = duckling_match["dim"]
-                        break
-                else:
-                    if self.duckling_processing_mode == "append":
-                        # Duckling will retrieve multiple entities, even if they overlap..
-                        # hence the append mode might add some noise to the found entities
-                        entities.append({
-                            "entity": duckling_match["dim"],
-                            "duckling": duckling_match["dim"],
-                            "value": duckling_match["value"]["value"],
-                            "start": duckling_match["start"],
-                            "end": duckling_match["end"],
-                        })
+            matches = self.duckling.parse(text)
+            relevant_matches = [match for match in matches if match["dim"] in self.duckling_dimensions]
+            for match in relevant_matches:
+                entity = {"start": match["start"],
+                          "end": match["end"],
+                          "value": match["value"]["value"],
+                          "entity": match["dim"]}
 
+                extracted.append(entity)
+
+        extracted = self.add_extractor_name(extracted)
         return {
-            "entities": entities
+            "entities": entities.extend(extracted)
         }
 
     @classmethod
-    def load(cls, duckling_processing_mode):
+    def load(cls, duckling_dimensions=DUCKLING_DIMENSIONS):
```

Comment:
`duckling_dimensions` should be persisted to the metadata in the persist call. Shouldn't take default arg here.